src/sandstorm/grain.capnp

```diff
@@ -739,7 +752,17 @@ struct RoleDef {
   # "default", then such sharing actions will be treated as having an empty permissions set (the
   # user can open the grain, but the grain is told that the user has no permissions).
   #
+  # This field should *not* be used to specify a default role hint for sharing!  For that, use
+  # `defaultSelection` immediately below.
+  #
   # See also `ViewSharingLink.RoleAssignment.none`, below.
+
+  defaultSelection @6 :Bool = false;
```

Comment:
When you do that, make sure to fall back to `default` if there is no `defaultSelected`.